example/Benchmark.cpp

```diff
@@ -0,0 +1,380 @@
+        SpinerEOSDependsRhoSieTransformable<ShiftTransform> eos_rs(sp5file, matid); //eventually make it SpinerEOSDependsRhoSie<NullTransfom>, for example
+	EOSPAC eos_ref(matid); 
+
+        //These vectors will store the compute time for each model for the 20 trials
+        std::vector<double> time_sie_eos_ref_list, time_sie_rt_list, time_sie_rs_list;
+        std::vector<double> time_P_eos_ref_T_list, time_P_rt_T_list, time_P_rs_T_list;
+        std::vector<double> time_P_eos_ref_sie_list, time_P_rt_sie_list, time_P_rs_sie_list;
+        std::vector<double> time_T_back_eos_ref_list, time_T_back_rt_list, time_T_back_rs_list;
+
+        //These matrices will be filled with the EOS values (overwriting for each trail)
+        //Then at the end of the program, we export the matrix to a .csv to later use for accuracy tests
+        std::vector<std::vector<Real>> sie_eos_ref(nRho, std::vector<Real>(nT)); 
+        std::vector<std::vector<Real>> sie_rt(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> sie_rs(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> P_eos_ref_T(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rt_T(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rs_T(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> P_eos_ref_sie(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rt_sie(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rs_sie(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> T_back_eos_ref(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> T_back_rt(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> T_back_rs(nRho, std::vector<Real>(nT));
+
+        // === Benchmark Loop ===
+        for (int rep = 0; rep < ntimes; ++rep) {
+                // Clear matrices
+                for (auto& row : sie_eos_ref) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : sie_rt) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : sie_rs) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : P_eos_ref_T) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rt_T) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rs_T) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : P_eos_ref_sie) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rt_sie) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rs_sie) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : T_back_eos_ref) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : T_back_rt) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : T_back_rs) std::fill(row.begin(), row.end(), 0.0);
```

Comment:
Minor quibble (again, not worth changing): since all matrices are the same size, this could just be one loop. The way you're doing it is fine though since this is just a test.